OpenRA.Mods.Common/UpdateRules/Rules/20221203/RemoveLegacyBridge.cs

```diff
@@ -0,0 +1,50 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class RemoveLegacyBridge : UpdateRule
+	{
+		public override string Name => "`LegacyBridgeHut` and `LegacyBridgeLayer` removed.";
+
+		public override string Description =>
+			"`LegacyBridgeHut` and `LegacyBridgeLayer` were removed, use `BridgeHut`\n" +
+			"and `BridgeLayer` instead. `NorthOffset` and `SouthOffset` were removed \n" +
+			"from `Bridge` and replaced by `NeighbourOffsets`.";
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			actorNode.RenameChildrenMatching("LegacyBridgeLayer", "BridgeLayer");
+
+			foreach (var actor in actorNode.ChildrenMatching("LegacyBridgeHut"))
+			{
+				actor.RenameKey("BridgeHut");
+
+				// WIP
+				actor.AddNode("NeighbourOffsets", new CPos[] { new CPos(0, 0) });
+			}
+
+			foreach (var actor in actorNode.ChildrenMatching("Bridge"))
+			{
+				actor.RemoveNodes("NorthOffset");
+				actor.RemoveNodes("SouthOffset");
+				actor.RemoveNodes("RepairPropagationDelay");
+
+				// WIP
```

Comment:
WIP?

Reply:
I dunno a way to automatically generate the offsets. I recon we may not even want to have offsets defined manually, rather scanning the whole map and looking for patterns

Reply:
Implemented the automatic system